### PR TITLE
tests: fixture records with reference text and identifier

### PR DESCRIPTION
**Pull request by ptamarit**
:heart: Thank you for your contribution!

Fixes https://github.com/zenodo/ops/issues/415

### Description

The test in `test_marcxml_serializer.py` has an assertion on `999C5$x` which make it look as if the reference identifier was serialized instead of the reference text.
This is due to the fact that the fake test data did not follow the real format (with a reference text, an identifier and a scheme).

This pull request makes the test data for references match with the reality and modifies affected assertions.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -548,7 +548,9 @@ def full_record(users):
             ],
             "references": [
                 {
-                    "reference": "0000 0001 1456 7559",
+                    "reference": "Nielsen et al,..",
+                    "identifier": "0000 0001 1456 7559",
+                    "scheme": "isni",
```
Comment (ptamarit) [inline tests/conftest.py:553]: Remark: this fixture was not affecting the MARCXML serializer test, but I changed it anyway so we're following the same format everywhere.

**Review by alejandromumo — APPROVED** [review]
left a comment for @ptamarit to decide on how to proceed with one of the changes

File: tests/conftest.py (hunk 1 above)
Comment (ptamarit) [reply tests/conftest.py]: OK nice.
What I just did is to remove the change to `full_record` and only keep the changes in `serializers/conftest.py`.
This way, you should be able to merge this PR and not have conflicts in the performance improvements branch.

Comment (ptamarit) [reply tests/conftest.py:553]: Sorry I was wrong, `full_record` is what is used in `test_marcxml_serializer_full_record`.
I'll mark this pull request as draft for now, and come back to it once you're done with the other more important pull requests.

Comment (ptamarit) [reply tests/conftest.py:553]: I confirm that following @alejandromumo's latest changes, the test in `test_marcxml_serializer.py` is not using `full_record_to_dict`. So now, the first comment I wrote is correct:

> Remark: this fixture was not affecting the MARCXML serializer test, but I changed it anyway so we're following the same format everywhere.

**Review by slint — APPROVED** [review]

**Pull request merged**